modules/Head/components/DynamicPreviewBranding.tsx

```diff
@@ -17,7 +17,18 @@ interface Props {
 
 export function DynamicPreviewBranding({ settings }: Props) {
     const searchParams = useSearchParams();
-    const parsedPreviewSettings = parsePreviewSearchParams(searchParams);
+    const searchParamsObject = useMemo(
+        () =>
+            Array.from(searchParams.entries()).reduce(
+                (result, [key, value]) => ({
+                    ...result,
+                    [key]: value,
+                }),
+                {},
+            ),
+        [searchParams],
+    );
```

Comment:
Can this be moved into `parsePreviewSearchParams()`?

Reply:
Maybe.. I wasn't sure if it's the best idea to make the `parsePreviewSearchParams` work with both plain objects as well as `URLSearchParams` interface since most of the time the input argument is a plain object.

We're only passing the `URLSearchParams` once in this place, so I thought it's okay to have the code here.